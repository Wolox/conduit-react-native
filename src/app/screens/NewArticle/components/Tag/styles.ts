import { StyleSheet } from 'react-native';
import { darkGrayMedium } from '@constants/colors';

const DEFAULT_SPACE = 10;
const DEFAULT_MARGIN = 5;

export default StyleSheet.create({
  tagContainer: {
    backgroundColor: darkGrayMedium,
    paddingHorizontal: DEFAULT_SPACE,
    marginVertical: DEFAULT_MARGIN,
    alignItems: 'center',
    borderRadius: 20,
    flexDirection: 'row',
    flexWrap: 'wrap',
    maxWidth: 200,
    height: 30
  },
  tagDelete: {
    paddingHorizontal: DEFAULT_SPACE,
    marginLeft: DEFAULT_MARGIN,
    alignItems: 'center',
    borderRadius: 100
  },
  removeItem: {
    marginLeft: DEFAULT_MARGIN,
    width: 20,
    borderRadius: DEFAULT_SPACE
  }
});
