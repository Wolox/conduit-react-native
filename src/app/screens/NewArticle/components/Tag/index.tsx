import React from 'react';
import { TouchableWithoutFeedback, View } from 'react-native';
import CustomText from '@components/CustomText';

import styles from './styles';

interface Props {
  text: string;
  index: number;
  onDeleteTag: (index: number) => void;
}

const MAX_LENGTH_WORD_TAG = 15;

function Tag({ text, onDeleteTag, index }: Props) {
  const textFormmated =
    text.length > MAX_LENGTH_WORD_TAG ? `${text.substring(0, MAX_LENGTH_WORD_TAG)}..` : text;

  return (
    <View style={styles.tagContainer}>
      <CustomText center white style={styles.text}>
        {textFormmated}
      </CustomText>
      <TouchableWithoutFeedback style={styles.tagDelete} onPress={() => onDeleteTag(index)}>
        <View style={styles.removeItem}>
          <CustomText bold white center>
            x
          </CustomText>
        </View>
      </TouchableWithoutFeedback>
    </View>
  );
}

export default Tag;
