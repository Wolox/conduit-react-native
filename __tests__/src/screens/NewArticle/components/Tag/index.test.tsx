import React from 'react';
import renderer from 'react-test-renderer';
import Tag from '@screens/NewArticle/components/Tag';

describe('Tag', () => {
  test('Tag Snapshot', () => {
    const onDeleteTag = (_index: number) => null;
    const item = { text: '', onDeleteTag, index: 0 };
    const tree = renderer.create(<Tag {...item} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
