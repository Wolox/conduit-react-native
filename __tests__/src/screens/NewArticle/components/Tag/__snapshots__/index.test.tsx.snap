// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Tag Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#818A91",
      "borderRadius": 20,
      "flexDirection": "row",
      "flexWrap": "wrap",
      "height": 30,
      "justifyContent": "center",
      "marginVertical": 5,
      "maxWidth": 200,
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#000",
          "fontFamily": "Nunito",
          "fontSize": 25.142857142857142,
          "fontStyle": "normal",
          "fontWeight": "400",
        },
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "color": "#FFF",
          },
        ],
        Object {
          "marginLeft": 5,
        },
      ]
    }
  >
    
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 10,
        "marginLeft": 5,
        "width": 20,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#000",
            "fontFamily": "Nunito",
            "fontSize": 25.142857142857142,
            "fontStyle": "normal",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#000",
              "fontFamily": "Nunito",
              "fontSize": 25.142857142857142,
              "fontStyle": "normal",
              "fontWeight": "700",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#FFF",
            },
          ],
          undefined,
        ]
      }
    >
      x
    </Text>
  </View>
</View>
`;
